### PR TITLE
User story3

### PR DESCRIPTION
**Pull request by NicolaiKopka**

### CODE REVIEW

File: backend/src/main/java/com/github/NicolaiKopka/users/AccountService.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class AccountService {
+
+    private final MyUserRepo myUserRepo;
+    private final PasswordEncoder encoder;
+    public MyUser registerUser(RegisterData registerData) {
+
+        if(registerData.getUsername().isBlank() || registerData.getPassword().isBlank() || registerData.getCheckPassword().isBlank()) {
```
Comment (gossie) [inline backend/src/main/java/com/github/NicolaiKopka/users/AccountService.java]: I would check username and the two passwords for null as well.

File: backend/src/main/java/com/github/NicolaiKopka/users/AccountService.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            throw new IllegalArgumentException("No empty fields allowed");
+        }
+        if(myUserRepo.findByUsername(registerData.getUsername()).isPresent()) {
```
Comment (gossie) [inline backend/src/main/java/com/github/NicolaiKopka/users/AccountService.java:23]: If two users try to create an user with the same username at the same time, both would end up in the database. Add a unique index for the field username in your `MyUser` class.

File: backend/src/main/java/com/github/NicolaiKopka/web_page_services/user_page/UserPageController.java [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import org.springframework.web.bind.annotation.*;
 
 @RestController
-@RequestMapping("/soundtracker/user")
+@RequestMapping("api/soundtracker/user")
 public class UserPageController {
 
     private UserPageService userPageService;
```
Comment (gossie) [inline backend/src/main/java/com/github/NicolaiKopka/web_page_services/user_page/UserPageController.java]: This field should probably be final.

File: frontend/src/Header/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import {NavLink} from "react-router-dom";
+
+export default function Header(){
+
+    return (
+        <div>
+            <NavLink to={"/"}><button>Main</button></NavLink>
+            {window.location.pathname !== "/login" &&
```
Comment (gossie) [inline frontend/src/Header/Header.tsx:8]: You can probably use the `useLocation`-hook instead.

**Review by gossie — CHANGES_REQUESTED** [review]

File: frontend/src/Header/Header.tsx (hunk 4 above)
Comment (NicolaiKopka) [reply frontend/src/Header/Header.tsx:8]: edited request changes

**Review by gossie — APPROVED** [review]

**Pull request merged**